core/templates/dev/head/services/GenerateContentIdService.ts

```diff
@@ -17,40 +17,47 @@
  * SubtitledHtml domain objects.
  */
 
-angular.module('oppia').factory('GenerateContentIdService', [
-  'COMPONENT_NAME_FEEDBACK', 'COMPONENT_NAME_HINT',
-  'COMPONENT_NAME_WORKED_EXAMPLE', function(
-      COMPONENT_NAME_FEEDBACK, COMPONENT_NAME_HINT,
-      COMPONENT_NAME_WORKED_EXAMPLE) {
-    var generateIdForComponent = function(existingComponentIds, componentName) {
-      var contentIdList = angular.copy(existingComponentIds);
-      var searchKey = componentName + '_';
-      var count = 0;
-      for (var contentId in contentIdList) {
-        if (contentIdList[contentId].indexOf(searchKey) === 0) {
-          var splitContentId = contentIdList[contentId].split('_');
-          var tempCount =
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { AppConstants } from 'app.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GenerateContentIdService {
+  generateIdForComponent(existingComponentIds: Object,
+      componentName: string): string {
+    let contentIdList = JSON.parse(JSON.stringify(existingComponentIds));
+    let searchKey = componentName + '_';
+    let count = 0;
```

Comment:
Do we need to define the type of the local (block scope) variable as well?

/cc @bansalnitish

Reply:
Yes, we should do this ideally. Tbh, there are so many variables, function arguments, function return types that adding types to each thing will definitely effect the pace. I think we need to discuss this once (Also, this should've been done during typescript migration ideally but that's fine).

What do you think @kevinlee12 @seanlip @Showtim3 @ankita240796?

Reply:
I strongly think we should do typing for variables and return types in public APIs, but I feel that typing for local variables can happen in a separate, later phase. That said, I would appreciate understanding what others' thoughts are on this, too. 

@kevinlee12 @Showtim3 @ankita240796 what do you think?

Reply:
Agreed, we can type variables inside functions later.

Reply:
Agreed! And yeah we are adding the args and the return type's. So we got it covered.

Reply:
Sounds good to me too!